File: app-java/src/main/java/net/mEmoZz/yts/java/ui/splash/SplashPresenterImpl.java
package net.mEmoZz.yts.java.ui.splash;

/**
 * Authored by Mohamed Fathy on 08 Mar, 2018.
 * Contact: muhamed.gendy@gmail.com
 */

class SplashPresenterImpl implements SplashPresenter {

  private SplashView splashView;

  SplashPresenterImpl(SplashView splashView) {
    this.splashView = splashView;
  }

  @Override public void onAttach() {
    splashView.openMainActivity();
  }

  @Override public void onDestroy() {
    splashView = null;
  }
}

File: app-kotlin/src/main/java/net/mEmoZz/yts/kotlin/ui/splash/SplashPresenterImpl.kt
package net.mEmoZz.yts.kotlin.ui.splash

/**
 * Authored by Mohamed Fathy on 08 Mar, 2018.
 * Contact: muhamed.gendy@gmail.com
 */

internal class SplashPresenterImpl(private var splashView: SplashView?) : SplashPresenter {

  override fun onAttach() {
    splashView!!.openMainActivity()
  }

  override fun onDestroy() {
    splashView = null
  }
}
